internal/pkg/deploy/pipeline.go

```diff
@@ -67,93 +70,100 @@ func (a *ArtifactBucket) Region() (string, error) {
 	return parsedArn.Region, nil
 }
 
-// Source defines the source of the artifacts to be built and deployed.
-type Source struct {
-	// The name of the source code provider. For example, "GitHub"
-	ProviderName string
+// GitHubSource defines the (GH) source of the artifacts to be built and deployed.
+type GitHubSource struct {
+	ProviderName                string
+	Branch                      string
+	RepositoryURL               string
+	PersonalAccessTokenSecretID string
+}
 
-	// Contains provider-specific configurations, such as:
-	// "repository": "aws/amazon-ecs-cli-v2"
-	// "githubPersonalAccessTokenSecretId": "heyyo"
-	Properties map[string]interface{}
+// CodeCommitSource defines the (CC) source of the artifacts to be built and deployed.
+type CodeCommitSource struct {
+	ProviderName  string
+	Branch        string
+	RepositoryURL string
 }
 
 // GitHubPersonalAccessTokenSecretID returns the ID of the secret in the
 // Secrets manager, which stores the GitHub Personal Access token if the
 // provider is "GitHub". Otherwise, it returns the detected provider.
-func (s *Source) GitHubPersonalAccessTokenSecretID() (string, error) {
-	// TODO type check if properties are GitHubProperties?
-	secretID, exists := s.Properties[manifest.GithubSecretIdKeyName]
-	if !exists {
+func (s *GitHubSource) GitHubPersonalAccessTokenSecretID() (string, error) {
+	if s.PersonalAccessTokenSecretID == "" {
 		return "", errors.New("the GitHub token secretID is not configured")
 	}
+	return s.PersonalAccessTokenSecretID, nil
+}
 
-	id, ok := secretID.(string)
-	if !ok {
-		return "", fmt.Errorf("unable to locate the GitHub token secretID from %v", secretID)
+// parseOwnerAndRepo parses the owner and repo name from the GH repo URL, which was formatted and assigned in cli/pipeline_init.go.
+func (s *GitHubSource) parseOwnerAndRepo() (string, string, error) {
```

Comment:
nit:
```suggestion
func (s *GitHubSource) parseOwnerAndRepo() (owner, repo string, error) {
```
to bring clarity into which string is for owner and which one for repo

Reply:
done